src/nccl_ofi_rdma.c

```diff
@@ -3316,6 +3307,73 @@ static int recv_close(nccl_net_ofi_recv_comm_t *recv_comm)
 	}
 
 	free(r_comm);
+
+exit:
+	return ret;
+}
+
+static int send_comm_destroy(nccl_net_ofi_rdma_send_comm_t *s_comm)
+{
+	int ret = 0;
+
+	/* Release connect response request if available */
+	if (s_comm->conn_resp_req) {
+		nccl_net_ofi_rdma_req_t *req = s_comm->conn_resp_req;
+		req->free(req, false);
+	}
+
+	/* Release request freelist */
+	ret = nccl_ofi_freelist_fini(s_comm->nccl_ofi_reqs_fl);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Call to nccl_ofi_freelist_fini failed: %d", ret);
+		goto exit;
+	}
+
+	if (!nccl_ofi_msgbuff_destroy(s_comm->msgbuff)) {
+		NCCL_OFI_WARN("Failed to destroy msgbuff (s_comm)");
+		ret = -EINVAL;
+		goto exit;
+	}
+
+	nccl_net_ofi_rdma_ep_t *ep = (nccl_net_ofi_rdma_ep_t *) s_comm->base.base.ep;
+	nccl_net_ofi_rdma_device_t *device = get_device_from_ep(ep);
+	set_comm(device, s_comm->local_comm_id, NULL);
+
+	/* Release communicator ID */
+	ret = nccl_ofi_idpool_free_id(device->comm_idpool, s_comm->local_comm_id);
+	if (OFI_UNLIKELY(ret != 0)) {
+		NCCL_OFI_WARN("Error freeing communicator ID %"PRIu32"", s_comm->local_comm_id);
+		goto exit;
+	}
+
+	/* Destroy domain */
+#if HAVE_NVTX_TRACING && NCCL_OFI_NVTX_TRACE_PER_COMM
+	for (int i = 0; i < NCCL_OFI_N_NVTX_DOMAIN_PER_COMM; ++i) {
+		nvtxDomainDestroy(s_comm->nvtx_domain[i]);
+	}
+#endif
+
+	free(s_comm);
+
+exit:
+	return ret;
```

Comment:
I know you're just copying, but there's really no good reason for the goto exit exception handling when there's no cleanup step.